lib/broadway_rabbitmq/producer.ex

```diff
@@ -159,7 +159,11 @@ defmodule BroadwayRabbitMQ.Producer do
       requeue: requeue?(config[:requeue], redelivered)
     }
 
-    message = %Message{data: payload, acknowledger: {__MODULE__, channel, ack_data}}
+    message = %Message{
+      data: payload,
+      metadata: meta,
```

Comment:
In order to get metadata in Amazon SQS, we made it opt-in. Should we do the same here? What are your thoughts @msaraiva?

Reply:
I agree. As far as I could see, we have at least the following attributes in the metadata:

```elixir
[
  :app_id,
  :cluster_id,
  :consumer_tag,
  :content_encoding,
  :content_type,
  :correlation_id,
  :delivery_tag,
  :exchange,
  :expiration,
  :headers,
  :message_id,
  :persistent,
  :priority,
  :redelivered,
  :reply_to,
  :routing_key,
  :timestamp,
  :type,
  :user_id
]
```
Probably just some of them like `routing_key`, `redelivered` and `headers` are actually important, so I would allow retriving any of them as long as it is explicitly using a `metadata` option, similar to `attributes` in the SQS client:

```
producers: [
    default: [
      module: {BroadwayRabbitMQ.Producer,
        queue_name: "my_queue",
        metadata: [:routing_key, :headers]
      }
    ]
  ]
```
@bdusauso WDYT? Can you implement this change? Let me know if you need any assistance.

Thanks!

Reply:
I like this idea as well, I find it clean and explicit.
I've done a basic implementation, can you please have a look at it and help me improve it ?

Thanks.